homeassistant/components/sentry/__init__.py

```diff
@@ -11,12 +12,68 @@
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers import config_validation as cv
 
-from .const import CONF_DSN, CONF_ENVIRONMENT, DOMAIN
+from .const import (
+    CONF_DSN,
+    CONF_ENVIRONMENT,
+    CONF_TRACING,
+    CONF_TRACING_SAMPLE_RATE,
+    DOMAIN,
+)
+
+
+def _get_integrations() -> list:
+    """Detect available Sentry integrations.
+
+    Return a list of enabled integrations based on modules available.
+    """
+    integrations = []
+
+    try:
+        import sqlalchemy  # NOQA
+        from sentry_sdk.integrations.sqlalchemy import SqlalchemyIntegration
+    except ImportError:
+        pass
+    else:
+        integrations.append(SqlalchemyIntegration())
+
+    try:
+        import aiohttp  # NOQA
+        from sentry_sdk.integrations.aiohttp import AioHttpIntegration
+    except ImportError:
+        pass
+    else:
+        integrations.append(AioHttpIntegration())
+
+    return integrations
+
+
+def sample_rate(value: Optional[Any]) -> float:
```

Comment:
We don't need this custom validator. Use the available validators either in `config_validation` helper or in the voluptuous package. We have this eg:
https://github.com/home-assistant/core/blob/1169ac568b9fdb87fbb30ead83fe441d5d3cef4f/homeassistant/helpers/config_validation.py#L86